example/src/screens/JoinRoom.tsx

```diff
@@ -0,0 +1,235 @@
+import { TextColors } from '@colors';
+import { BackgroundAnimation } from '@components/BackgroundAnimation';
+import { Modal } from '@components/Modal';
+import { TextInput } from '@components/TextInput';
+import { Typo } from '@components/Typo';
+import { StandardButton } from '@components/buttons/StandardButton';
+import { RootStack } from '@model/NavigationTypes';
+import { useHeaderHeight } from '@react-navigation/elements';
+import type { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { useCardAnimation } from '@react-navigation/stack';
+import {
+  checkIfUrl,
+  extractRoomNameFromUrl,
+  shouldEnableRoomButton,
+} from '@utils';
+import React, { useEffect, useState, useCallback, useRef } from 'react';
+import {
+  View,
+  StyleSheet,
+  KeyboardAvoidingView,
+  Platform,
+  PermissionsAndroid,
+  ScrollView,
+} from 'react-native';
+import { useVideoroomState } from 'src/VideoroomContext';
+
+type Props = NativeStackScreenProps<RootStack, 'JoinRoom'>;
+type GoBackAction = Readonly<{
+  type: string;
+  payload?: object | undefined;
+  source?: string | undefined;
+  target?: string | undefined;
+}>;
+
+export const JoinRoom = ({ navigation, route }: Props) => {
+  const height = useHeaderHeight();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const modalAction = useRef<GoBackAction>();
+  const { roomName, setRoomName, username, setUsername } = useVideoroomState();
+  const [isRoomNameInputEditable, setIsRoomNameInputEditable] = useState(true);
+  const { next, current } = useCardAnimation();
+
+  useEffect(() => {
+    if (route.params?.roomName) {
+      setRoomName(route.params?.roomName);
+      setIsRoomNameInputEditable(false);
+    }
+  }, []);
+
+  useEffect(() => {
```

Comment:
You're repeating code here. I propose creating a separate component like `<DiscardModal />` that handles the navigation stuff too. You can use `useNavigation` hook for accessing the navigation object outside the screen component.